File: win7gadget/gadget/gadget/Content/rpc.js
rpc = function() { }

rpc.login = function(url, login, password, callback, callbackErr) {
    xmlrpc(
        url + "/rpc/xmlrpc",
        "jira1.login",
        [login, password],
        function(ret) {
            callback(ret);
        },
        function(err) {
            callbackErr(err);
        },
        function() { }
    );
}

rpc.getprojects = function(url, token, callback, callbackErr) {
    xmlrpc(
        url + "/rpc/xmlrpc",
        "jira1.getProjectsNoSchemes",
        [token],
        function(ret) {
            callback(ret);
        },
        function(err) {
            callbackErr(err);
        },
        function() {
        }
    );
}

File: win7gadget/gadget/gadget/rpc.cs
using System.Runtime.CompilerServices;

namespace gadget {

    public delegate void LoginCallback(string token);
    public delegate void GetProjectsCallback(object result);
    public delegate void CallbackError(string error);

    [Imported]
    [IgnoreNamespace]
    public class rpc {
        public static void login(string url, string login, string password, LoginCallback cb, CallbackError errorCb) { }
        public static void getprojects(string url, string token, GetProjectsCallback cb, CallbackError errorCb) { }
    }
}
